javalib/src/main/scala/java/util/concurrent/ConcurrentSkipListSet.scala

```diff
@@ -0,0 +1,423 @@
+
+      def hasNext(): Boolean = iter.hasNext
+
+      def next(): E = {
+        last = Some(iter.next().inner)
+        last.get
+      }
+
+      def remove(): Unit = {
+        if (last.isEmpty) {
+          throw new IllegalStateException()
+        } else {
+          last.foreach(self.remove(_))
+          last = None
+        }
+      }
+    }
+  }
+
+  def descendingSet(): NavigableSet[E] = {
+    val descSetFun = () => {
+      val inverseOrdering =
+        new Ordering[Box[E]] {
+         def compare(a: Box[E], b: Box[E]) = self.ordering.compare(b.inner,a.inner)
+        }
+      val retSet = new mutable.TreeSet[Box[E]]()(inverseOrdering)
+      retSet.addAll(inner)
+      retSet
+    }
+    new ConcurrentSkipListSet.NavigableView(this, descSetFun)
+  }
+
+  def first(): E =
+    if (inner.isEmpty) throw new NoSuchElementException()
+    else inner.head.inner
+
+  def last(): E =
+    if (inner.isEmpty) throw new NoSuchElementException()
+    else inner.last.inner
+
+  def ceiling(e: E): E =
+    tailSet(e, true).headOption.getOrElse(null.asInstanceOf[E])
+
+  def floor(e: E): E =
+    headSet(e, true).lastOption.getOrElse(null.asInstanceOf[E])
+
+  def higher(e: E): E =
+    tailSet(e, false).headOption.getOrElse(null.asInstanceOf[E])
+
+  def lower(e: E): E =
+    headSet(e, false).lastOption.getOrElse(null.asInstanceOf[E])
+
+  def pollFirst(): E = {
+    val polled = inner.headOption
+    if (polled.isDefined) {
+      val elem = polled.get.inner
+      remove(elem)
+      elem
+    } else null.asInstanceOf[E]
+  }
+
+  def pollLast(): E = {
+    val polled = inner.lastOption
+    if (polled.isDefined) {
+      val elem = polled.get.inner
+      remove(elem)
+      elem
+    } else null.asInstanceOf[E]
+  }
+
+  override def remove(o: Any): Boolean =
+    if (o == null) throw new NullPointerException()
+    else inner.remove(Box(o.asInstanceOf[E]))
+
+  override def removeAll(c: Collection[_]): Boolean = {
+    val iter = c.iterator()
+    var changed = false
+    while (iter.hasNext)
+      changed = inner.remove(iter.next) || changed
+
+    changed
+  }
+
+  def size(): Int =
+    inner.size
+
+  def headSet(toElement: E): NavigableSet[E] =
+    headSet(toElement, false)
+
+  def headSet(toElement: E, inclusive: Boolean): NavigableSet[E] = {
+    val headSetFun = () => {
+      val boxed = Box(toElement)
+      val hs = new mutable.TreeSet[Box[E]]()
+      hs ++= inner.until(boxed)
+      if (inclusive && inner.contains(boxed))
+        hs += boxed
+      hs
+    }
+    new ConcurrentSkipListSet.NavigableView(this, headSetFun)
+  }
+
+  def tailSet(fromElement: E): NavigableSet[E] =
+    tailSet(fromElement, true)
+
+  def tailSet(fromElement: E, inclusive: Boolean): NavigableSet[E] =  {
+    val tailSetFun = () => {
+      val boxed = Box(fromElement)
+      val ts = new mutable.TreeSet[Box[E]]()
+      ts ++= inner.from(boxed)
+      if (!inclusive && inner.contains(boxed))
+        ts remove boxed
```

Comment:
Style: use `ts.remove(boxed)`. We use infix notation only for symbolic operators, or if the argument is a lambda with `{}`. I think there are a couple more of these.